src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Interop/UnsafeNativeMethodsWindow.cs

```diff
@@ -0,0 +1,344 @@
+    /// <param name="window">Window to check.</param>
+    public static bool IsWindowHasLegacyMica(Window window) =>
+        GetHandle(window, out IntPtr windowHandle) && IsWindowHasLegacyMica(windowHandle);
+
+    /// <summary>
+    /// Tries to determine whether the provided handle has applied legacy Mica effect.
+    /// </summary>
+    /// <param name="handle">Window handle.</param>
+    public static bool IsWindowHasLegacyMica(IntPtr handle)
+    {
+        if (!NativeMethods.IsWindow(handle))
+        {
+            return false;
+        }
+
+        var pvAttribute = 0x0;
+
+        _ = Dwmapi.DwmGetWindowAttribute(
+            handle,
+            Dwmapi.DWMWINDOWATTRIBUTE.DWMWA_MICA_EFFECT,
+            ref pvAttribute,
+            Marshal.SizeOf(typeof(int))
+        );
+
+        return pvAttribute == 0x1;
+    }
+
+    /// <summary>
+    /// Tries to apply legacy Mica effect for the selected <see cref="Window"/>.
+    /// </summary>
+    /// <param name="window">The window to which the effect is to be applied.</param>
+    /// <returns><see langword="true"/> if invocation of native Windows function succeeds.</returns>
+    public static bool ApplyWindowLegacyMicaEffect(Window window) =>
+        GetHandle(window, out IntPtr windowHandle) && ApplyWindowLegacyMicaEffect(windowHandle);
+
+    /// <summary>
+    /// Tries to apply legacy Mica effect for the selected <see cref="Window"/>.
+    /// </summary>
+    /// <param name="handle">Window handle.</param>
+    /// <returns><see langword="true"/> if invocation of native Windows function succeeds.</returns>
+    public static bool ApplyWindowLegacyMicaEffect(IntPtr handle)
+    {
+        var backdropPvAttribute = 0x1; //Enable
+
+        _ = Dwmapi.DwmSetWindowAttribute(
+            handle,
+            Dwmapi.DWMWINDOWATTRIBUTE.DWMWA_MICA_EFFECT,
+            ref backdropPvAttribute,
+            Marshal.SizeOf(typeof(int))
+        );
+
+        return true;
+    }
+
+    /// <summary>
+    /// Tries to apply legacy Acrylic effect for the selected <see cref="Window"/>.
+    /// </summary>
+    /// <param name="window">The window to which the effect is to be applied.</param>
+    /// <returns><see langword="true"/> if invocation of native Windows function succeeds.</returns>
+    public static bool ApplyWindowLegacyAcrylicEffect(Window window) =>
+        GetHandle(window, out IntPtr windowHandle) && ApplyWindowLegacyAcrylicEffect(windowHandle);
+
+    /// <summary>
+    /// Tries to apply legacy Acrylic effect for the selected <see cref="Window"/>.
+    /// </summary>
+    /// <param name="handle">Window handle</param>
+    /// <returns><see langword="true"/> if invocation of native Windows function succeeds.</returns>
+    public static bool ApplyWindowLegacyAcrylicEffect(IntPtr handle)
+    {
+        var accentPolicy = new ACCENT_POLICY
+        {
+            nAccentState = ACCENT_STATE.ACCENT_ENABLE_ACRYLICBLURBEHIND,
+            nColor = 0x990000 & 0xFFFFFF
+        };
```

Comment:
This will introduce severe performance problems on older supported versions of Windows. Like I mentioned in an earlier comment, you'll need to consider how this effort/code will work across all supported versions of Windows.

Reply:
Again, the method wasn't called / used for applying the acrylic background. Have removed all its references.